src/BasketBundle/Form/PaymentType.php

```diff
@@ -99,7 +100,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
         $sub = $builder->create('paymentMethod', $choiceType, array(
             'expanded' => true,
-            'choice_list' => new SimpleChoiceList($choices),
+            'choice_list' => function (Options $opts, $previousValue) use ($choices) {
```

Comment:
@ZRazor I checked you code...
do you think that I need to create a 'ChoiceLoader' for every Type (for example: DeliveryChoiceLoader', ...) or I can create one that works for all ? because the code will be always the same... 
or, another solution is to create an abstract class with the code and many class as many loader I need...